source/kraken/apply_disruption.cpp

```diff
@@ -200,21 +200,66 @@ type::ValidityPattern compute_base_disrupted_vp(const std::vector<boost::posix_t
+        if (network == nullptr) {
+            LOG4CPLUS_INFO(logger, "Rejecting impact (" << impact.uri << "): network_id (" << impact.network_id
+                                                        << ") provided is absent");
+            return nullptr;
+        }
+    } else {
+        network = pt_data.get_or_create_network("network:additional_service", "additional service");
+    }
+
+    nt::CommercialMode* comm_mode = nullptr;
+    if (!impact.commercial_mode_id.empty()) {
+        comm_mode = pt_data.get_commercial_mode(impact.commercial_mode_id);
+        if (comm_mode == nullptr) {
+            LOG4CPLUS_INFO(logger, "Rejecting impact (" << impact.uri << "): commercial_mode_id ("
+                                                        << impact.commercial_mode_id << ") provided is absent");
+            return nullptr;
+        }
+    } else {
+        comm_mode = pt_data.get_or_create_commercial_mode("commercial_mode:additional_service", "additional service");
+    }
 
     // We get the first and last stop_area to create route and line
     const auto& st_depart = impact.aux_info.stop_times.front();
     const auto& sa_depart = st_depart.stop_time.stop_point->stop_area;
     const auto& st_arrival = impact.aux_info.stop_times.back();
     const auto& sa_arrival = st_arrival.stop_time.stop_point->stop_area;
-    std::string line_uri = "line:" + sa_depart->uri + "_" + sa_arrival->uri;
-    std::string line_name = sa_depart->name + " - " + sa_arrival->name;
-    std::string route_uri = "route:" + sa_depart->uri + "_" + sa_arrival->uri;
-    std::string route_name = sa_depart->name + " - " + sa_arrival->name;
-    nt::Line* line = pt_data.get_or_create_line(line_uri, line_name, network, comm_mode);
-    nt::Route* route = pt_data.get_or_create_route(route_uri, route_name, line, sa_arrival, "forward");
+    nt::Line* line = nullptr;
+    if (!impact.line_id.empty()) {
+        line = pt_data.get_line(impact.line_id);
+        if (line == nullptr) {
+            LOG4CPLUS_INFO(logger, "Rejecting impact (" << impact.uri << "): line_id (" << impact.line_id
+                                                        << ") provided is absent");
+            return nullptr;
+        }
+    } else {
+        std::string line_uri = "line:" + sa_depart->uri + "_" + sa_arrival->uri;
+        std::string line_name = sa_depart->name + " - " + sa_arrival->name;
+        line = pt_data.get_or_create_line(line_uri, line_name, network, comm_mode);
+    }
+
+    std::string route_uri = "";
+    std::string route_name = "";
+    std::string direction_type = "";
+    if (!impact.route_id.empty()) {
+        route_uri = impact.route_id;
+        route_name = "Additional service";
+        direction_type = "outbound";
+    } else {
+        route_uri = "route:" + sa_depart->uri + "_" + sa_arrival->uri;
+        route_name = sa_depart->name + " - " + sa_arrival->name;
+        direction_type = "forward";
```

Comment:
```suggestion
        direction_type = "outbound";
```

Reply:
Be sure of name, direction type .. as explain in the doc

Reply:
OK for me as i found: 
Requirement : same behavior than before when values are empty (existing tests should not change).